opentracing-mock/src/main/java/io/opentracing/mock/MockSpan.java

```diff
@@ -262,4 +276,12 @@ static long nextId() {
     static long nowMicros() {
         return System.currentTimeMillis() * 1000;
     }
+
+    private synchronized void finishedCheck(String message) {
+        if (finished) {
+            RuntimeException ex = new IllegalStateException(message);
+            errors.add(ex);
```

Comment:
In some cases you cannot rely on the fact that a thrown exception by `MockSpan` would fail/error a test. E.g. in some cases an exception would be just logged by a container and everything in tests would look normal. Therefore I would prefer to have an explicit way to check if something non-compliant with spec happened.